django_select2/forms.py

```diff
@@ -399,15 +398,18 @@ def filter_queryset(self, request, term, queryset=None, **dependent_fields):
         select = Q()
         term = term.replace("\t", " ")
         term = term.replace("\n", " ")
-        for t in [t for t in term.split(" ") if not t == ""]:
-            select &= reduce(
-                lambda x, y: x | Q(**{y: t}),
-                search_fields[1:],
-                Q(**{search_fields[0]: t}),
-            )
+
+        for field in search_fields:
+            field_select = Q()
+            if "contains" in field:
+                for t in [t for t in term.split(" ") if not t == ""]:
+                    field_select &= Q(**{field: t})
+            else:
+                field_select = Q(**{field: term})
+            select |= field_select
```

Comment:
Do you actually need to handle the cases differently? It would still work, if you return both results, right?

```suggestion
        for field in search_fields:
            field_select = Q(**{field: term})
            for word in filter(None, term.split(" ")):
                field_select |= Q(**{field: t})
                
            select |= field_select
```

Reply:
Prior to swapping the `&` for `|`, if searching for "Spa ce" you'd end up with:
```
WHERE (
  "testapp_genre"."title" LIKE %Spa ce% ESCAPE '\' AND
  "testapp_genre"."title" LIKE %Spa% ESCAPE '\' AND
  "testapp_genre"."title" LIKE %ce% ESCAPE '\'
)
```
This fails `test_filter_queryset`

After swapping the AND for OR you get:
```
WHERE (
  "testapp_genre"."title" LIKE %Spa ce% ESCAPE '\' OR
  "testapp_genre"."title" LIKE %Spa% ESCAPE '\' OR
  "testapp_genre"."title" LIKE %ce% ESCAPE '\'
)
```
But this is actually functionally different to what the old code does. [AND](https://github.com/codingjoe/django-select2/blob/master/django_select2/forms.py#L403) is used for the individual parts of the search term, meaning with `search_fields=["title__icontains"]` you get:
```
WHERE (
  "testapp_genre"."title" LIKE %Spa% ESCAPE '\' AND
  "testapp_genre"."title" LIKE %ce% ESCAPE '\'
)
```
I'm not sure whether AND or OR is prefered in this case.

The OR in the old code is actually applied to the different `search_fields`, so if we do the same search but with `search_fields=["title__icontains", "title__istartswith"]` you get:
```
WHERE (
  (
    "testapp_genre"."title" LIKE %Spa% ESCAPE '\' OR
    "testapp_genre"."title" LIKE Spa% ESCAPE '\'
  ) AND (
    "testapp_genre"."title" LIKE %ce% ESCAPE '\' OR
    "testapp_genre"."title" LIKE ce% ESCAPE '\'
  )
)
```
I'm not sure that is intuitive either, but might have been done by design at some point. What I'd expect, and what the current changes do, would be:
```
WHERE (
  (
    "testapp_genre"."title" LIKE %Spa% ESCAPE '\' AND
    "testapp_genre"."title" LIKE %ce% ESCAPE '\'
  ) OR 
  "testapp_genre"."title" LIKE Spa ce% ESCAPE '\'
)
```